### ISSUE TITLE
Inconsistent video splitting

### ISSUE THREAD

**Issue by alegaballo** [Breakthrough/PySceneDetect#29]
I'm having some troubles trying to save the splits of a video.
If I run the following command 
`scenedetect -i myvideo.mp4 -d content -t 30`
I get the following output
```
...
[PySceneDetect] Processing complete, found 76 scenes in video.
...
```

If I try to save the output as .mkv `scenedetect -i myvideo.mp4 -d content -t 30 -o myvideo-scene.mkv` I only get 25 scenes.
If I try to save only the .csv file `scenedetect -i myvideo.mp4 -d content -t 30 -o scenes.csv` it looks like the program is trying to write the actual video splits in a csv files because in the end I get again 25 csv files

**Comment by mohannad57** [comment]
I'm not sure about the inconsistency but I know that on version 0.4 this was changed but the documentations are still. 

"•specifying  -o OUTPUT_FILE.mkv  will now automatically split the input video, generating a new video clip for each detected scene in sequence, starting with  OUTPUT_FILE-001.mkv 
•CSV file output is now specified with the  -co  /  --csv-output  option (note, used to be  -o  in versions of PySceneDetect < 0.4)"
source: https://github.com/Breakthrough/PySceneDetect/releases

**Comment by alegaballo** [comment]
Oh okay, I didn't notice that, thank you. Now the .csv file is correct but still the number of scenes if I try to use -o scene.mkv is different

**Comment by mohannad57** [comment]
That part works for me just fine. so I will leave it for the experts as I'm very new to all this. :) sorry

**Comment by Breakthrough** [comment]
Hmm interesting... Using the new command-line options (`-o` for the video output, `-co` for the CSV output), could you provide the full output you get now?  Of the scenes that *are* written to disk, are they cut at the appropriate place (near where the timecode in the CSV output is)?

**Comment by alegaballo** [comment]
Sorry for being this late, I was quite busy. Try with  [this video](https://www.youtube.com/watch?v=KQtB3-NMfnU) with the default configuration:
 `scenedetect -i video.mp4 -c scenes.csv -o out.mp4`
the number of scenes is correct but their duration is different from the one described in the csv

**Comment by Breakthrough** [comment]
Hello all;

This has been fixed in the most recent version of PySceneDetect (v0.5-beta).  The stable v0.5 release should be out sometime this month.

The issue is that when just splitting videos *without* re-encoding, they can only be split in certain cases on keyframe boundaries.  This also means that sometimes, when several scenes are very small and close together, they may be concatenated together, or not generated properly.

To fix this, I have integrated PySceneDetect v0.5 with `ffmpeg` to allow re-encoding the output video.  To accomplish this in v0.5, you would use the `split-video` command with the `-p/--precise` option:

    scenedetect -i input_video.mp4 -o output_dir detect-content split-video -p

You can also save thumbnail images now via the `save-images` command.

    scenedetect -i input_video.mp4 -o output_dir detect-content split-video -p save-images

I have just tested this on v0.5, and indeed, some output files were unreadable without either the `-p` (precise/re-encode) or `-m` (use mkvmerge) flags.  Feel free to test the v0.5-beta if you have a chance and provide some feedback.

Obviously using the `-p` flag is much slower than just copying the streams, but it is significantly more accurate in terms of where the output cuts are placed.  You can also override the codec options, e.g.:

    scenedetect -i input_video.mp4 -o output_dir detect-content split-video -f "-c:v libx264 -c:a aac"
